src/engine/strat_engine/backstore/blockdevmgr.rs

```diff
@@ -196,13 +189,17 @@ impl BlockDevMgr {
         let devices = process_and_verify_devices(pool_uuid, &current_uuids, paths)?;
 
         let encryption_info = self.encryption_info();
-        if encryption_info.is_some() && !self.has_valid_passphrase() {
+        if encryption_info.is_encrypted()
+            && !self.can_unlock(
+                encryption_info.key_description.is_some(),
+                encryption_info.clevis_info.is_some(),
+            )
+        {
             return Err(StratisError::Engine(
                 ErrorEnum::Invalid,
-                "The key associated with the current registered key description \
-                was not able to unlock an existing encrypted device; check that \
-                the same key is in the keyring that was used to create the encrypted \
-                pool"
+                "Neither the keyring nor Clevis could be used to unlock the device; \
```

Comment:
Given that this is simulating an unlock, I don't think what I have here is incorrect. Can you explain a little bit more? The only time the passphrase is used is for encryption operations, not just general operations.

Reply:
"the device" is a device belonging to the pool that the person who sees this text probably won't know about. I had "encryption operations" originally, but thought that was redundant, since Clevis and "kernel keyring" both imply encryption operations. But  "encryption operations" probably does make it clearer.